File: cpp/alerts/request_dropped.cpp
#include "./request_dropped.h"

namespace ltrs {

int request_dropped_alert_get_block_index(lt::request_dropped_alert* alert) {
    return alert->block_index;
}

PieceIndex request_dropped_alert_get_piece_index(lt::request_dropped_alert* alert) {
    return PieceIndex { .inner = static_cast<int32_t>(alert->piece_index) };
}

} // namespace ltrs

File: src/ffi/alerts/request_dropped.rs
#[cxx::bridge(namespace = "ltrs")]
pub(crate) mod ffi {
    #[namespace = "libtorrent"]
    extern "C++" {
        type request_dropped_alert = crate::ffi::ffi::request_dropped_alert;
    }

    unsafe extern "C++" {
        include!("cpp/alerts/request_dropped.h");
        type PieceIndex = crate::ffi::ffi::PieceIndex;
        unsafe fn request_dropped_alert_get_block_index(alert: *mut request_dropped_alert) -> i32;
        unsafe fn request_dropped_alert_get_piece_index(
            alert: *mut request_dropped_alert,
        ) -> PieceIndex;
    }
}
